packages/lesswrong/components/posts/dialogues/DialogueEditorFeedback.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { getSiteUrl, registerComponent } from '../../../lib/vulcan-lib';
+import Button from '@material-ui/core/Button';
+import { useCurrentUser } from '../../common/withUser';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    padding: 20
+  },
+  button: {
+    fontSize: 16,
+    textTransform: 'none',
+    backgroundColor: theme.palette.buttons.startReadingButtonBackground,
+  },
+  feedbackRow: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+  }
+});
+
+type clickState = 'unclicked' | 'clickedAndItWorked' | 'clickedAndItDidntWork'
+
+const feedbackButtonClicked = async (user: UsersCurrent | null, post: PostsEdit, setBeenClicked: Dispatch<SetStateAction<clickState>>) => {
```

Comment:
By convention we define such functions inside the component, if they rely on e.g. `setState` calls (and just make them a closure over the `setState` call.  I realize this is terribly impure but absent thought don't want to start fragmenting the codebase into multiple patterns